python/ray/autoscaler/_private/fake_multi_node/node_provider.py

```diff
@@ -344,6 +344,9 @@ def create_node_with_resources_and_labels(
     def _create_node_with_resources_and_labels(
         self, node_config, tags, count, resources, labels
     ):
+        # This function calls `pop`. To avoid side effects, we make a
```

Comment:
Without this, the first node in the scaling request will be created correctly. However, the following nodes will be created without CPU/GPU because their resources have already been popped from resources.